UPGRADING.md

```diff
@@ -0,0 +1,117 @@
+- `api.tx.balances.transfer(BOB, '0x3039')` - value specied as a hex
+- `api.tx.balances.transfer(BOB, new BN(12345))` - value specied as a [BN](https://github.com/indutny/bn.js/)
+
+Internally the API will take the input and convert the value into a `Balance`, serialize it using the SCALE codec and transfer it to the node. In some cases users would construct the `Balance` type manually, by importing the class and calling `new` on it. This last approach has now been removed, and where classes are still available (limited reach), discouraged.
+
+First the rationale behind this - in all cases Substrate is very flexible, so while Polkadot (and the Substrate base), define `type Balance = u128`, this can be different between chains. (This also applies to the majority of built-in supported types). As such, type construction should be done via the actual registered types.
+
+```js
+// this is applicable everywhere, import the type creator, using the registry
+import { createType } from '@polkadot/types';
+
+// construct the Balance, of type Balance (type is inferred and available with TS)
+const value = createType('Balance', 12345);
+
+// use value here as you normally would
+...
+```
+
+The impact of this will be noticable, if you have been importing the old-style type classes from `@polkadot/types`, those imports are not available anymore. For creation, just pass everything through the `createType`.
+
+If a TypeScript user, you can find the updated type (it is a type definition only, not a class), under `@polkadot/types/interfaces`. To do type casting, using interfaces -
+
+```js
+// import the TypeScript runtime interfaces we wish to use
+import { Balance, Hash } from '@polkadot/types/interfaces';
+
+// import the primitives we wish to use
+import { createType, Compact, Vec, u32 } from '@polkadot/types';
+
+// define an interface we want to use inside our code
+interface MyProps {
+  balance: Compact<Balance>;
+  changes: Vec<Hash>;
+  counter?: u32;
+}
+
+// assign something to this structure
+const props = {
+  balance: createType('Compact<Balance>', 12345),
+  changes: createType('Vec<Hash>', []) // empty for now
+};
+```
+
+### Type definitions
+
+One of the major painpoints in working with a custom Substrate node is the definition of types to cater for chains. There are 2 approaches: definiting types via a JSON format or extending your own classes in TypeScript (or JS) and injecting these. For the latter category, there are some impacts in the way you define these.
+
+If using JSON definitions, nothing changes, your types are still defined as -
+
+```json
+{
+  "MyStruct": {
+    "balance": "Compact<Balance>",
+    "values": "Vec<AccountId>",
+    "counter": "u32"
+  }
+}
+```
+
+For he definition of any structures using the Substrate specific types as classes, some adjustments are needed. Since the base modules types are now not available in classes, however it is needed for definitions, the following approach is encouraged -
```

Comment:
```suggestion
For the definition of any structures using the Substrate specific types as classes, some adjustments are needed. Since the base modules types are now not available in classes, however it is needed for definitions, the following approach is encouraged -
```